src/commands/synthetics/reporters/junit.ts

```diff
@@ -392,8 +499,43 @@ export class JUnitReporter implements Reporter {
           {$: {name: 'type', value: test.type}},
         ].filter((prop) => prop.$.value),
       },
-      testcase: [],
+      skipped: [],
       warning: [],
     }
   }
+
+  private getTestCaseStats(result: Result): TestCaseStats {
+    let stepsStats: TestCaseStats[] = []
+    if ('stepDetails' in result.result) {
+      // It's a browser test.
+      stepsStats = result.result.stepDetails
+        .map((step) => {
+          if (!step.subTestStepDetails) {
+            return [step]
+          }
+
+          return [step, ...step.subTestStepDetails]
+        })
+        .reduce((acc, val) => acc.concat(val), [])
+        .map(this.getBrowserStepStats)
+    } else if ('steps' in result.result) {
+      // It's an multistep API test
+      stepsStats = result.result.steps.map(this.getApiStepStats)
+    } else {
+      stepsStats = [this.getApiStepStats(result.result)]
+    }
+
+    const stats = getDefaultTestCaseStats()
+
+    for (const stepStats of stepsStats) {
+      stats.steps_count += stepStats.steps_count
+      stats.steps_errors += stepStats.steps_errors
+      stats.steps_failures += stepStats.steps_failures
+      stats.steps_skipped += stepStats.steps_skipped
+      stats.steps_allowfailures += stepStats.steps_allowfailures
+      stats.steps_warnings += stepStats.steps_warnings
+    }
+
+    return stats
```

Comment:
Not a requested change, but an example using `reduce` to avoid using a for loop and some mutation (just for the sake of sharing this kind of knowledge):

```ts
return stepsStats.reduce(
  (stats, stepStats) => ({
    ...stats,
    steps_count: stats.steps_count + stepStats.steps_count,
    // etc.
  }), getDefaultTestCaseStats())
```